### ISSUE TITLE
createReducer allows setting unknown fields to new state

### ISSUE THREAD

**Issue by sregg** [omichelsen/redux-promise-middleware-actions#17]
![image](https://user-images.githubusercontent.com/1479283/72473317-6d8fc100-37b4-11ea-824d-f7d3be2eb31f.png)

See line 17 `nane: action.payload.name`, TS is nit complaining about that I typed `nane` instead of `name`.

If I add the type manually:

![image](https://user-images.githubusercontent.com/1479283/72473415-a0d25000-37b4-11ea-90e6-d6b2a2b88cec.png)

Then I see the error fine.

Is there a way to have the error without the manual type?

Thanks

**Comment by sregg** [comment]
Here's the code to repro:

```
import { createAction, createReducer } from 'redux-promise-middleware-actions';

interface IState {
  name: string;
}

const INITIAL_STATE: IState = {
  name: '',
};

const setName = createAction('SET_NAME', (name: string) => ({ name }));

export const NameReducer = createReducer(INITIAL_STATE, handleAction => [
  // intro screen
  handleAction(setName, (state, action) => ({
    ...state,
    nane: action.payload.name,
  })),
]);

```

**Comment by omichelsen** [comment]
Good point, I will try and add this. If you have a solution a PR is of course very welcome :)

Currently it would actually give you a warning if you only set the `name` property, but because you are spreading the state (`...state`) it has already set `name` (to the old value) and will not warn you because you are allowed to extend the object with a new `nane` prop.
